File: teaching/lesson2/students.rb
require "./common"
require "./other"

class Student
  include Common

  def initialize(name, scores)
    @name = name
    @scores = scores
  end
  
  def getName
    return @name
  end

  def getAverageScore
    sum = 0.0
    @scores.each {
      |score|
      sum += score
    }
    return sum / @scores.length
  end
end

def compareM(x, y)
 return y.getAverageScore <=> x.getAverageScore
end 

def main()
puts "Ranking of students:"
s0 = Student.new("David L", [90, 86, 60, 10])
s1 = Student.new("Jovani K", [92, 84, 83, 12])
s2 = Student.new("Alexa O", [43, 23, 22, 5])
students = [s0, s1, s2]
students.sort! {|x, y| compareM(x, y)}
symbol = :hello
d = {:hello => "name = #{s0.getName}"}
p d[:hello]
students.each {
  |s|
  puts "%s \t (%.2f)\n" % [s.getName, s.getAverageScore]
  d[s.getName] = s
}

puts d["David L"].getName
s0.greeting
Other::hello
puts match("$hello", "hello there")
end

main()

File: teaching/lesson2/students.js
var $ = {};

$.set = function(id, html) {
  var el = document.getElementById(id);
  el.innerHTML = html;
}

var Student = function(name, grades) {
  this.name = name;
  this.grades = grades;
}

Student.prototype.average_score = function() {
  var sum = 0;
  for (var i = 0; i < this.grades.length; i++) {
    sum += this.grades[i];
  }
  return sum / this.grades.length;
}

var ctrl = {};

ctrl.main = function() {
  var s1 = new Student("David L", [90, 86, 60, 10]);
  var s2 = new Student('Jovani K', [92, 84, 83, 12]);
  var s3 = new Student('Alexa O', [43, 23, 22, 5]);
  var students = [s1, s2, s3];
  var str = "";
  students.sort(function(a, b) {return a.average_score() < b.average_score();});
  for (var i = 0; i < students.length; i++) {
    var s = students[i];
    str += "<div>"
    str += s.name + " \t(" + s.average_score() + ")";
    str += "</div>";
  }
  $.set('force', str);
}

ctrl.main();